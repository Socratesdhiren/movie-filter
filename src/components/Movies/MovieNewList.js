import React, { useState, useEffect } from "react";
import { Row, Pagination, Input, Spin } from "antd";
import MovieCard from "./MovieCard";
import {getMoviesList} from './API/MoviesAPI';

const { Search } = Input;

const MoviesList = () => {
  const [movies, setMovies] = useState([]);
  const [totalPage, setTotalPage] = useState();
  const [currentPage, setCurrentPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [searchText, setSearchText] = useState("fast");

  const fetchMoviesList = async(page, searchValue) => {
    setLoading(true);
   const response = await getMoviesList(searchValue, page);
   if(response){
    setMovies(response?.data?.Search || []);
    setTotalPage(response?.data?.totalResults);
   }
 
   setLoading(false)
  };

  const handlePageChange = (value) => {
    setCurrentPage(value);
  };

  const onSearch = (Value) => {
    setSearchText(Value);
  };

  useEffect(() => {
    fetchMoviesList(currentPage, searchText);
  }, [currentPage, searchText]);

  return (
    <Spin spinning={loading} delay={200}>
      <Row>
        <Pagination
          defaultCurrent={currentPage}
          total={totalPage}
          onChange={handlePageChange}
          showSizeChanger={false}
        />

        <Search
          placeholder="input search text"
          onSearch={onSearch}
          defaultValue={searchText}
          enterButton
          style={{ width: "400px", marginLeft: "200px" }}
        />
      </Row>

      <Row>
        {movies &&
          movies?.length &&
          movies.map((movie) => (
            <div key={movie?.imdbID}>
              <MovieCard movie={movie} />
            </div>
          ))}
      </Row>
    </Spin>
  );
};
export default MoviesList;
