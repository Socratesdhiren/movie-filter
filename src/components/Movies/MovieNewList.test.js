import renderer from "react-test-renderer";

import MovieNewList from "./MovieNewList";

describe("movies list", () => {
  
//   test("renders MovieNewList", () => {
//     render(<MovieNewList />);
//   });
  
});

describe("SnapShot matches", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<MovieNewList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
