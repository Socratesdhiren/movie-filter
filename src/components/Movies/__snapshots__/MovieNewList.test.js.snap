// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot matches renders correctly 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div
    className="ant-spin-container"
  >
    <div
      className="ant-row"
      style={Object {}}
    >
      <ul
        className="ant-pagination"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
      <span
        className="ant-input-group-wrapper ant-input-search ant-input-search-with-button"
        style={
          Object {
            "marginLeft": "200px",
            "width": "400px",
          }
        }
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            hidden={null}
            onBlur={[Function]}
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="input search text"
            style={null}
            type="text"
            value="fast"
          />
          <span
            className="ant-input-group-addon"
          >
            <button
              className="ant-btn ant-btn-primary ant-input-search-button"
              onClick={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              <span
                aria-label="search"
                className="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </span>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      0
    </div>
  </div>
</div>
`;
